definitions/procedures/backup/prepare_directory.rb

```diff
@@ -6,18 +6,29 @@ class PrepareDirectory < ForemanMaintain::Procedure
       param :backup_dir, 'Directory where to backup to', :required => true
       param :preserve_dir, 'Directory where to backup to', :flag => true
       param :incremental_dir, 'Changes since specified backup only'
+      param :online_backup, 'Select for online backup', :flag => true, :default => false
     end
 
+    # rubocop:disable Metrics/MethodLength
     def run
       puts "Creating backup folder #{@backup_dir}"
 
       unless @preserve_dir
         FileUtils.mkdir_p @backup_dir
         FileUtils.chmod_R 0o770, @backup_dir
-      end
 
-      if feature(:instance).postgresql_local? && !@preserve_dir
-        FileUtils.chown_R(nil, 'postgres', @backup_dir)
+        if feature(:instance).postgresql_local? && @online_backup
+          begin
+            FileUtils.chown_R(nil, 'postgres', @backup_dir)
+          rescue Errno::EPERM
+            warn_msg = <<~MSG
+              #{@backup_dir} could not be made readable by the 'postgres' user.
+              This won't affect the backup procedure, but you have to ensure that
+              the 'postgres' user can read the data during restore.
+            MSG
+            set_status(:warning, warn_msg)
+          end
+        end
       end
 
       FileUtils.rm(Dir.glob(File.join(@backup_dir, '.*.snar'))) if @preserve_dir
```

Comment:
Should this be in the `else` part of `unless @preserve_dir`?

Reply:
Maybe. But honestly I didn't want to touch too much of the code, just the parts that I actually wanted to change.

Also, I am not really sure why this is guarded by `if @preserve_dir` anyway.
As when it's `false`, the `@backup_dir` will be fresh and empty and the `.glob` will just return `[]`.